src/org/webonise/multithreading/producerconsumer/Producer.java

```diff
@@ -0,0 +1,37 @@
+package org.webonise.multithreading.producerconsumer;
+
+import java.util.Queue;
+
+public class Producer implements Runnable {
+    private final Queue<Integer> queue;
+
+    Producer(Queue<Integer> queue){
+        this.queue = queue;
+    }
+
+    @Override
+    public void run() {
+        int resource = 0;
+
+        while(true){
+            putResourceIntoQueue(resource++);
+        }
+    }
+
+    private void putResourceIntoQueue(int resource){
+
+        String threadName = Thread.currentThread().getName();
+
+        synchronized (queue){
+            queue.add(resource);
+            System.out.println(threadName + " adding resource : " + resource);
+            queue.notifyAll();
+
+            try{
+                queue.wait();
```

Comment:
The logic behind this wait was to let producer wait until it receive notifications from consumers that they have fetched the resource from queue and the producer can now add more resources to queue. I have updated it right now with timed wait to prevent infinite waiting by producer.